server/src/main/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutor.java

```diff
@@ -1070,6 +1089,11 @@ private static String pathFromRevision(long revision) {
     @Override
     protected <T> CompletableFuture<T> doExecute(Command<T> command) throws Exception {
         final CompletableFuture<T> future = new CompletableFuture<>();
+        ExecutorService executor = this.executor;
+        if (command instanceof UpdateServerStatusCommand) {
+            // Use a separate executor because `this.executor()` could be stopped while executing the command.
+            executor = ForkJoinPool.commonPool();
```

Comment:
ditto. only when the replicating is false?

Reply:
Replied at https://github.com/line/centraldogma/pull/921#discussion_r1571865798